src/nvim/option.c

```diff
@@ -3822,8 +3865,25 @@ set_bool_option (
    * End of handling side effects for bool options.
    */
 
+  // after handling side effects, call autocommand
+
   options[opt_idx].flags |= P_WAS_SET;
 
+  if (!starting) {
+    char_u buf_old[2], buf_new[2], buf_type[7];
+    vim_snprintf((char *)buf_old, 2, "%d", old_value ? true: false);
+    vim_snprintf((char *)buf_new, 2, "%d", value ? true: false);
```

Comment:
> %d expects >=2 bytes (system-dependent) int

Is that true? 

>|conversion specifier|description|
|:----------|:-------------:|
|d,i | The **int** argument is converted to signed decimal in the style _[−]dddd_. The precision specifies the minimum number of digits to appear; if the value being converted can be represented in fewer digits, it is expanded with leading zeros. The default precision is 1. The result of converting a zero value with a precision of zero is no characters.|

and

> ### integer promotion
> * The rank of **_Bool** shall be less than the rank of all other standard integer types.
> * The following may be used in an expression wherever an **int** or **unsigned int** may
be used:
>   - An object or expression with an integer type whose integer conversion rank is less
than or equal to the rank of **int** and **unsigned int**.
> * If an **int** can represent all values of the original type, the value is converted to an **int**;
otherwise, it is converted to an **unsigned int**. These are called the _integer
promotions_.
> *  The integer promotions preserve value including sign.

cf : http://www.open-std.org/JTC1/SC22/WG14/www/docs/n1124.pdf

Implicit(correct):
```c
vim_snprintf((char *)buf_new, 2, "%d", value ? true: false)
```

explicit(also correct):
```c
vim_snprintf((char *)buf_new, 2, "%d", value ? (int)true: (int)false)
```